pyresample/spherical_utils.py

```diff
@@ -0,0 +1,194 @@
+of an area is covered by the onboard scanning instrument(s).
+"""
+
+import itertools
+
+
+class GetNonOverlapUnionsBaseClass():
+    """Base class to get the smallest set of union objects that does not overlap.
+
+    The objects are here Python sets of integers - but are abstracts for
+    geometrical shapes on a sphere.
+
+    """
+
+    def __init__(self, geom_objects):
+        self.geometries = geom_objects
+        self.geom_numbers = range(len(geom_objects))
+
+        self._geoms = dict(enumerate(geom_objects))
+
+    def merge(self):
+        """Merge all overlapping objects (sets or polygons).
+
+        """
+        all_unions = self._merge_unions(self._geoms)
+        check_keys_int_or_tuple(all_unions)
+
+        self._geoms = {}
+        for key, value in all_unions.items():
+            if isinstance(key, int):
+                self._geoms[key] = value
+            else:
+                newkey = merge_tuples(key)
+                self._geoms[newkey] = value
+
+    def get_polygons(self):
+        """Get a list of all non-overlapping polygon unions."""
+        return list(self._geoms.values())
+
+    def get_ids(self):
+        """Get a list of identifiers identifying the gemoetry objects in each polygon union."""
+        return list(self._geoms.keys())
+
+    def _overlaps(self, set1, set2):
+        """Do the two sets overlap each other (have anything in common)?"""
+        if set1 != set1.difference(set2):
+            return True
+
+        return False
+
+    def _find_union_pair(self, geoms):
+        """From a set of geometries find a pair that overlaps.
+
+        *geoms* is here expected to be a numbered dict with SphPolygon
+        polygons. If no pair of two polygons overlap (that is the union
+        returns something different from None or False) then return None.
+
+        Strictly the geometries/objects does not need to be a SphPolygon. The
+        only requirement is that it has a union method with the same behaviour.
+
+        """
+
+        if len(geoms) == 1:
+            return None
+
+        for id_, komb_pair in zip(itertools.combinations(geoms.keys(), 2),
+                                  itertools.combinations(geoms.values(), 2)):
+            if self._overlaps(komb_pair[0], komb_pair[1]):
+                return id_, komb_pair[0].union(komb_pair[1])
+
+        return None
+
+    def _merge_unions(self, geoms):
+        """Merge all overlapping geometry unions.
+
+        Go through the dictionary of geometries and merge overlapping ones until
+        there is no change anymore. That is, the input dict of geometries is the
+        same as the output:
+
+        """
+
+        retv = self._find_union_pair(geoms)
+        if retv is None:
+            return geoms
+
+        cc_poly = geoms.copy()
+
+        for idx in [0, 1]:
+            del cc_poly[retv[0][idx]]
+        cc_poly[retv[0]] = retv[1]
+
+        return self._merge_unions(cc_poly)
```

Comment:
If I understand correctly, the current implementation does the union twice: first to see if there is an overlap, and then to actually perform the union. This could be very slow for large polygons...

Reply:
Yeah, you might have a point here! The use case that I am focusing on is a set of satellite/sensor overpasses over a geographical region within a time interval of a few minutes to 2-3 hours. The longer the time interval and the more satellites the more polygons we get. So far it has been okay time wise.